src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarReader.cs

```diff
@@ -361,83 +362,60 @@ private bool TryGetNextEntryHeader(out TarHeader header, bool copyData)
                 header = mainHeader;
             }
 
-            // Common fields should always acquire a value
-            Debug.Assert(header._name != null);
-            Debug.Assert(header._linkName != null);
-
-            // Initialize non-common string fields if necessary
-            header._magic ??= string.Empty;
-            header._version ??= string.Empty;
-            header._gName ??= string.Empty;
-            header._uName ??= string.Empty;
-            header._prefix ??= string.Empty;
-
-            return true;
+            return header != null;
```

Comment:
Isn't this always `return true;` ? I don't believe header is ever `null` here, is it?

Reply:
Good point. Actually, this method's signature should be made to look like the async version (return `TarHeader?` instead of `bool`), so that I can de-duplicate code easier in the next PR.